hamocc/mo_ocprod.F90

```diff
@@ -729,6 +778,29 @@ subroutine ocprod(kpie,kpje,kpke,kbnd,pdlxp,pdlyp,pddpo,omask,ptho,pi_ph,psao,pp
               ocetra(i,j,k,ioxygen) = ocetra(i,j,k,ioxygen) - ro2utammo*remin
               remin_aerob(i,j,k)  = remin_aerob(i,j,k)+remin*rnit ! kmol/NH4/dtb - remin to NH4 from various sources
             endif
+            if (use_river2omip) then
+              ocetra(i,j,k,iphosph) = ocetra(i,j,k,iphosph) + tdoclc_deg + tdochc_deg
+              if (.not. use_extNcycle) then
+                ocetra(i,j,k,iano3)   = ocetra(i,j,k,iano3)   + tdoclc_deg*rnit_tdoclc             &
+                                      &                       + tdochc_deg*rnit_tdochc
+                ocetra(i,j,k,ialkali) = ocetra(i,j,k,ialkali) - (rnit_tdoclc+1.)*tdoclc_deg        &
+                                      &                       - (rnit_tdochc+1.)*tdochc_deg
+                ocetra(i,j,k,ioxygen) = ocetra(i,j,k,ioxygen) - tdoclc_deg*ro2ut_tdoclc            &
```

Comment:
Hi @TimotheeBrgs , rate dependent, but this can lead to negative O2 concentrations (i.e. when the O2 consumption due to remin is larger then the available oxygen).